### DISCUSSION TITLE
기초프로젝트 1주차 팀회고

### DISCUSSION THREAD

**Discussion by yeji0512** [codeit-bootcamp-frontend/2-Weekly-Mission#311] [회고]
# 왜 회고를 해야하나요?
원활한 팀 프로젝트 진행을 위해, 회고가 좋은 도구가 되어줄거예요. 
매주 팀 단위로 회고를 하며 어떤 점이 좋았고, 어떤 점이 아쉬웠고, 아쉬웠던 부분은 어떻게 개선해볼지 함께 이야기해보면 
그 다음주에는 더 나은 프로젝트를 진행할 수 있을거예요. 회고를 통해 함께 성장해보아요 🔥 

# 회고 방법
![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/144666341/b1f1cc83-7d24-4156-910d-3fc9b6bb7555)
회고는 **“KPT 회고”** 방식으로 진행해볼거예요. 아래 방법대로 회고해주시면 됩니다. 
- 팀원들과 모여 그 주의 팀 프로젝트 활동을 돌아보며, '좋았던 점(Keep) / 아쉬웠던 점(Problem) / 시도해볼 것(Try)'으로 항목을 나누어 회고해주세요.
    - Keep: 현재 우리팀이 잘 하고 있어서 계속 유지하면 좋을 것 같은 부분
    - Problem: 현재 우리팀이 가지고 있는 문제
    - Try: Problem에 적은 것을 해결하기 위해 우리 팀이 다음에 시도해볼만한 것 (혹은 Keep 중에서도 더 개선하거나 시도해보고 싶은 것)
- [KPT 회고 피그마 템플릿](https://www.figma.com/file/0NHIhlM9vhYZDrtZa7nJn8/%ED%9A%8C%EA%B3%A0-%ED%85%9C%ED%94%8C%EB%A6%BF?type=whiteboard&node-id=0%3A1&t=p6vXDVEjQntHKaNP-1)을 활용해서, 팀 회고를 진행하셔도 됩니다. 단 템플릿 사용시, 팀별로 복제해서 사용해주세요. 꼭 템플릿을 사용하지 않아도 되니, 팀별로 편한 방식으로 회고를 진행해주세요. 
- **이번주 일요일까지, 팀 회고한 내용을 정리하여 댓글로 올려주세요.** 
(댓글에 예시 회고를 올려드릴테니, 참고해주세요. 피그마 템플릿으로 진행하셨다면 캡처본을 올려주셔도 돼요.)

# 참고 글
- [KPT 회고란 무엇인가?](https://brunch.co.kr/@jinha0802/35)
- [팀문화의 탄생](https://techblog.woowahan.com/2677/)

**Comment by yeji0512** [comment]
# 😎 0팀 (예시)
### Keep 
- 미팅 시간 약속을 모두 잘 지켜준다. 
- 프로젝트 과정에서 기술 고민을 함께 하는 것이 서로의 학습에 도움이 된다. 
- 새로운 제안을 했을 때, 모두 개방적으로 의견을 들어주어서 고맙다. 
- 위워크에 모여서 하다보니, 프로젝트 진행이 빠르게 잘 이루어진다. 

### Problem
- 커밋 메시지가 명시적이지 않아서, 서로 어떤 작업을 했는지 파악하기 힘들었다. 
- 급하게 코드를 작성하느라, 폴더 구조가 이상하다. 
- 서로를 너무 배려하느라, 의견을 더 많이 나누지 못한다. 

### Try 
- 협업을 위해, 커밋 메시지를 더 구체적으로 작성한다. (커밋 메시지 규칙 정하기) 
- 폴더 구조를 다시 논의하여 폴더 구조 규칙을 만들고, 규칙에 맞추어 파일을 배치한다. 
- 각자 하루에 한번 이상 의견을 낸다.

**Comment by 9ummy9** [comment]
> # ✨10팀✨
> ### Keep
> * 팀미팅 시간에 모두 성실히 참여했다.
> * 팀프로젝트를 통해 깃허브를 더 잘 이해하게 됐다.
> * 혼자 공부하는 것 보다 팀프로젝트로 함께 배워나가는 것이 학습에 도움이 됐다. 
> * 내가 생각하지 못했던 점을 팀원들이 알려줘서 집단지성의 힘을 느꼈다.
> 
> ### Problem
> * 공통 컴포넌트의 범위를 정하는 것이 가장 어려웠다. 다음 프로젝트를 하게 된다면 초기분석에서 공통 컴포넌트의 분류 및 정의를 더 자세히 내려야할 것 같다.
> * 내일까지 기한을 정했을 때 팀미팅 전까지인지, 자정까지인지 서로 생각하는 기준이 달랐다. 좀더 명확하게 기한을 말하기로 했다. 
> 
> ### Try
> * 2차 스프린트 기간에 대한 아이디어를 더 적극적으로 냈으면 좋겠다.
> * 최대한 빠르게 맡은 기능을 구현해냈음 좋겠다. (파이팅!)
> * Problem에 있었던 문제를 다음 프로젝트에서는 경험하지 않았으면 좋겠다.

![20180405154005_xzikivui](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148167964/5e0c353f-8bd6-40cb-b3c0-cd249195f00a)

**Comment by lhv0829** [comment]
# 👀5팀

### Keep
- 미팅 시간 약속을 모두 잘 지켜준다
- 각자 하고 싶은 부분이 명확해서 역할 배분이 쉬웠다
- 문제를 직면했을 때 먼저 해결해보기 위해 시도하는 것이 좋았다
- 의견을 제시했을 때 의견을 잘 들어주고 반영도 잘 되서 고맙다
- 모두 열의가 넘치셔서 하루에 회의를 두번 진행했는데도 잘 참여해주셔서 고맙다

### Problem
- 초반에 코드 컨벤션이랑 PR 형식같은 것들이 정해져있지 않아서 좀 알아보기 힘들었다
- 오류가 났는데도 제대로 확인하지 않고 PR을 올려서 문제가 생겼다
- 초반에 코드리뷰를 진행하지 않고 문제점을 말하기가 좀 어려워서 더 힘들었다
- API 명세를 잘 확인했으면 더 좋았을 것 같다
- 질문의 내용을 명확히 설명하지 못해 팀원들이 어떤 질문인지 이해하지 못하는 소통의 오류가 생겼다

### Try
- 협업을 위해 PR을 더 명확하게 작성한다
- PR올리기 전에 오류(merge된 부분 pull해서 합쳤을 때 잘 작동하는지 확인)나 불필요한 주석이 있는지 확인하고 올린다
- 팀원들이 이해할 수 있도록 질문의 내용을 명확히 정리하도록 노력한다

**Comment by jisunipark** [comment]
# 🔥8팀

![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148641571/e96c8028-e320-4eb7-85b5-6cd90fb9cbbd)

**Comment by rladngus133** [comment]
# :poop: 1팀

## Keep (우리팀이 잘 하고있는부분)
- **우현** : 1팀의 경우 작업방식을 페이지단위별로 했는데 각 각 맞춰진 일정에 늦지않게 그 날 구현,개발 할 것을 다 개발해내는 모습이 멋졌습니다. 프로젝트가 쉬운 내용은 아니여서 조금 늦을수도 있다고 생각했지만 다들 포기하지않고 맞춰진 시간내에 해결하는 모습이 좋왔습니다 ! 

- **윤수** : 개발을 하며 각 페이지에 대한 구현과정에서 서로 이 부분을 고치면 좋을것같다 등등의 얘기도 많이했었는데 다 긍정적으로 받아들이며 소통을 하는 부분이 좋은것 같았습니다. 전체적으로 팀의 분위기가 긍정적이라서 편안하게 할 수 있었던것 같았습니다. 

- **부근** : 기능개발 과정에서 어려운 부분이 꽤 있었다고 생각하는데 각자 어려운 기능을 개발시 빠르게 포기하지않고 끝까지 도전하여 결국 완성시키는 모습이 좋았던것 같습니다.

- **윤경** : 팀에서 소통이 잘 이루어지면서 유독 좋았던 부분은 추가기능에 대한 아이디어가 팀원들간에 끊이지 않는점이 좋왔습니다. 기능개발을 하며 이렇게 해볼수도 있구나 하며 참고도 되어 더욱 좋은것 같았습니다.

- **지수** : 소통이 원할하게 잘 됩니다. 페이지 기능작업중 문제가 발생하는 경우가 많았는데 문제 발생시 Discord를 통하여 소통하면 굉장히 빠르게 소통하여 기능구현하는대에 있어 원할하게 진행이 가능했던것 같습니다. 1팀짱~ 

## Problem (우리팀 문제)
- 다 자기가 못한다고 생각함... 🫠
- PR을 보내는 기준이 잘 안정해져 있다.
- 라이브러리를 필요할때마다 설치해서 충돌이 많이 일어난다.

## Try
- 각자 자신감 가지기
- 초기셋팅단계에서 조금더 열심히 소통해서 라이브러리등등 다 정하고 가기

# 1팀 파이팅~~~
![](https://velog.velcdn.com/images/rladngus133/post/48f2b167-f58f-45ed-a7d9-e9d9d93b0d74/image.gif)

**Comment by ys1008p** [comment]
7팀

  Keep
 - 미팅 시간 약속을 모두 잘 지켜준다.
 - 프로젝트 과정에서 기술 고민을 함께 하는 것이 서로의 학습에 도움이 된다.
 - 새로운 제안을 했을 때, 모두 개방적으로 의견을 들어주어서 고맙다.
 - 위워크에 모여서 하다보니, 프로젝트 진행이 빠르게 잘 이루어진다.

Problem
 - 커밋 메시지가 명시적이지 않아서, 서로 어떤 작업을 했는지 파악하기 힘들었다.
 - 급하게 코드를 작성하느라, 폴더 구조가 이상하다.
 - 서로를 너무 배려하느라, 의견을 더 많이 나누지 못한다.

Try
 - 협업을 위해, 커밋 메시지를 더 구체적으로 작성한다. (커밋 메시지 규칙 정하기)
 - 폴더 구조를 다시 논의하여 폴더 구조 규칙을 만들고, 규칙에 맞추어 파일을 배치한다.
 - 각자 하루에 한번 이상 의견을 낸다.

**Comment by grapefruit13** [comment]
# 2팀

<img width="825" alt="image" src="https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/92169354/f56a662b-e2ce-4279-a46a-7a82ee5e161a">

**Comment by juncastle97** [comment]
## 9️⃣ 팀

⭐️ Keep
- 각자 맡았던 파트를 기간 안에 잘 완료했다.
- 더 원활한 의사소통을 위해 위워크에 매일 나온다.
- 프로젝트 진행 과정을 잘 기록하고 있다.
- R&R 분배를 세세하게 나눠서 프로젝트 진행이 편했다.
- 팀 프로젝트 하면서 git에 대한 이해도가 올라갔다.

🤢 Problem
- 스크럼 시간이 너무 길어진다.
- 필요없는 주석 지우기.
- 서로 생각하는 바를 명확하게 전달하지 못했다.

🏃🏻 Try
- 자기 의견을 조금 더 강하게 주장하기(근거가 있을 때).
- PR 올리기 전에 오류, 콘솔창 확인하기 올리기.
- 추가 기능 구현을 노려본다.

![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/137983583/8094c5d6-9ef7-4bbf-8b13-ac1803c3446a)

**Comment by dkile** [comment]
# 😎 11팀

### Keep

- 코드 리뷰를 열심히 한다.
- 다들 맡은 바를 열심히 해주신다.
- 소통이 빠르고 활발하다.
- 유연한 마감 기한. 워라밸 보장. 상황 공유 잘 됨.
- github을 잘 이용하는 것 같다. wiki, projects 등등. 브랜치 관리도 잘 됨.

### Problem

- 맡은 바를 너무 열심히 하는 나머지 리뷰 속도가 조금 늦다.
- 질문을 잘 안한다.
- PR 크기가 애매하다.
- 실력 편차가 있는 것 같다. 몸에 좋은 거 같이 좀 먹읍시다. 너무 안락하다.
- 개인이 결정하기에 애매한 부분이 많다.

### Try

- 코드 리뷰를 조금만 더 빨리… 어푸르브…
    - 어떻게?
        - pr 크기를 줄이자.
        - 사전에 요구사항 분석을 좀 더 확실히 하자
        - 승인 기준을 명확히 하자.
- 1시간을 고민했는데 모르겠다. 디코에 질문을 올리자.
- 몸에 좋은 건 2차 프로젝트 때 주도적으로 해보자.
- 애매한 부분이 발생할 때마다 기록하고 팀 회의 때 공유하자.

**Comment by LeeJuyoung-design** [comment]
## **😎 6팀**

Keep

- 모두 코어 타임 맞춰서 코딩하는 것이 좋았어요.
- 코어타임 덕분에 모르는 것을 바로바로 질문할 수 있는 시간이 생겼어요.
- 깃허브에 다들 익숙하지 않을텐데, 제안하는대로 잘 해줘서 좋았어요.
- 팀 프로젝트로 인해 익숙하지 않았던 깃허브를 잘 사용하게 되어서 좋았습니다.
- **다들 친절하셔서 좋았어요.**


Problem

- PR을 하고 매일 한번씩 main 에 merge 작업을 하려고 하는데, 작업이 꽤나 귀찮아요. action을 사용할 줄 알았다면 좋았을 것 같아요.
- 깃허브 관련해서 여러가지 배우는 것도 많지만 아직 잘 안되는 것도 많아요. ㅠㅠ
- PR 코드리뷰가 빠르게 되지 않는 것 같아요.
- Global Style을 처음에 정할 걸! 후회 중이에요. (격하게 공감!)


Try

- action을 배워서 써보고 싶어요.
- 커밋은 컴포넌트 단위로 합시다.
- live share로 여태 짠 코드 한번 소개하기
- **커밋을 잊지 말기**
- 내가 구현한 기능을 사용하는 방법을 기록해야 할 것 같아요. (남도 사용해야 하니까)
- 이 웹 사용해서 안드로이드 앱도 만들어보고 싶어요.

**Comment by muyahho** [comment]
3팀🙏
<img width="746" alt="스크린샷 2023-12-17 13 49 34" src="https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/88885994/617c849a-48d8-460a-bf01-bf611e3de8bc">

**Comment by SeanKim-416** [comment]
# 🔥12팀🔥

## Keep
### 팀 내 소통이 활발하다.
- 모든 팀원이 위워크에 매일 나온다.
- 매일 미팅을 통해 각자의 진행상황을 잘 공유한다.
- 매일 진행상황을 팀 메세지를 통해 정리 및 공유하여 현재 위치를 파악할 수 있도록 했다.
- 각자 작업 방식이나, 작업 속도 등에 대한 다양성을 이해해줬다.

- 아이디어 제안을 적극적으로 한다.
- 제안된 아이디어를 긍정적으로 받아들이고 수렴하려 했다.

## Problem
### 협업 경험의 부족
- 깃을 다루는게 익숙하지 않아 충돌이나 에러가 종종 생겼다.
- 자신있게 프로젝트를 이끌 수 있는 역량은 부족했다.

### 개인 역량에 대한 인식
- 본인 실력에 대한 자신감이 없다.
- 본인의 작업속도를 정확하게 파악하지 못해 예상 작업시간과 실제 작업시간이 차이가 날 때가 있었다.
## Try
- 스스로의 실력에 대한 자신감을 가지자.
- 변수, 함수 네이밍을 서로가 이해하기 쉽게 직관적으로 고쳐보자.
- 커밋 메세지를 좀더 이해하기 쉽게 작성하자.
- 꼭 pull 하고 push하자!!

**Comment by sobokki** [comment]
# 😂4팀🤦‍♀️🤦‍♀️🤦‍♀️🤦‍♂️👏👍👌
## Keep

1. 주기적으로 만남을 가지고 소통을 열심히 했어요.
2. 약속한 기간 내에 각자의 분량을 완성했어요.
3. 막히는 부분이 있으면 서로 도와주며 같이 완성해 나갔어요.
4. 적극적인 태도로 매일 팀회의를 이어나가며, 부족한 부분에 대해 함께 논의하고 해결 방안을 결정했어요. 

## Problem
협업 경험이 부족해 깃을 다루는 능력, 용어 부분에 능숙하지 못했습니다. 
PR 하는 법이나 컴포넌트 이름을 정하는 규칙을 잘 정하지 못했습니다. 
## Try
위축되지 않고 더 노력하자
우울한 시간에 코드 공부하자 
그만둘 거 아니면 찡찡거리지 않고 어떻게든 구현하기 
공동체 정신을 더 가져보며 나는 혼자가 아니라는 마인드 장착하기

**Comment by lhv0829** [comment]
# 👀5팀(2주차)

### Keep
- 미팅 시간 약속을 모두 잘 지켜준다.
- 문제를 직면했을 때 먼저 해결해보기 위해 시도하는 것이 좋았다.
- 의견을 제시했을 때 의견을 잘 들어주고 반영도 잘 되서 고맙다.
- 회의가 길어져도 다 같이 열심히 참여해줘서 좋았다.

### Problem
- 프로젝트를 계획할 때 브랜치 전략이나 todolist를 명확하게 하지 않고 바로 코딩을 시작해 나중에 관리할 때 힘들었다.
- 코드리뷰를 하지 않아서 나중에 수정하기가 힘들었다.
- 디테일한 부분을 신경쓰지 못해서 아쉽다.


### Try
- 프로젝트 관리를 더 신경쓴다.
- 코드 리뷰를 좀 더 꼼꼼히 하는게 좋겠다.
- 디테일한 부분을 마지막까지 신경쓴다.
